driver/src/main/java/org/neo4j/driver/internal/messaging/v57/MessageReaderV57.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [https://neo4j.com]
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver.internal.messaging.v57;
+
+import java.io.IOException;
+import java.net.ProtocolException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.neo4j.driver.Value;
+import org.neo4j.driver.Values;
+import org.neo4j.driver.internal.messaging.GqlError;
+import org.neo4j.driver.internal.messaging.ResponseMessageHandler;
+import org.neo4j.driver.internal.messaging.v5.MessageReaderV5;
+import org.neo4j.driver.internal.packstream.PackInput;
+import org.neo4j.driver.types.Type;
+import org.neo4j.driver.types.TypeSystem;
+
+public class MessageReaderV57 extends MessageReaderV5 {
+    private static final Type STRING_TYPE = TypeSystem.getDefault().STRING();
+    private static final Type MAP_TYPE = TypeSystem.getDefault().MAP();
+
+    public MessageReaderV57(PackInput input) {
+        super(input);
+    }
+
+    @Override
+    protected void unpackFailureMessage(ResponseMessageHandler output) throws IOException {
+        var params = unpacker.unpackMap();
+        var gqlError = unpackGqlError(params);
+        output.handleFailureMessage(gqlError);
+    }
+
+    private GqlError unpackGqlError(Map<String, Value> params) throws IOException {
+        var gqlStatus = params.get("gql_status").asString();
+        var statusDescription = params.get("status_description").asString();
+        var code = params.get("neo4j_code").asString();
```

Comment:
This value, `neo4j_code`, is not present in `causes`. Since GqlCauses is something new we add, we won't need to keep the backwards compatibility here and have the code. We can also define the causes as GqlException and extend the Neo4jException from there. 

Does it make sense?

Reply:
Yeah, I am aware they won't have the `neo4j_code`. I will update that bit. They will be just `Neo4jException` objects with `N/A` code to avoid breaking updates.